Realm/RLMSyncConfiguration.mm

```diff
@@ -183,8 +184,10 @@ - (instancetype)initWithUser:(RLMUser *)user
                     break;
             }
 
-            RLMSyncManager *manager = [app syncManager];
-            auto errorHandler = manager.errorHandler;
+            RLMSyncErrorReportingBlock errorHandler;
+            @autoreleasepool {
```

Comment:
Is there a need for `@autoreleasepool` here?

Reply:
Accessing a weak pointer autoreleases the value. This code's running on the sync worker thread, so letting an autoreleased value escape from the function could result in it living a very long time. I don't think it'd lead to a retain cycle or the crash being fixed here, but minimizing how long we're holding a strong reference to the sync manager seems the most likely to avoid problems.